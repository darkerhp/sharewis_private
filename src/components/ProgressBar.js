import React from 'react';
import ReactNative from 'react-native';

const { PropTypes } = React;
const { StyleSheet, View } = ReactNative;

const styles = StyleSheet.create({
  progressView: {
    flexDirection: 'row',
    height: 18,
    backgroundColor: '#cdcdcd',
    borderColor: '#cdcdcd',
    borderWidth: 1,
  },
  currentProgress: {
    backgroundColor: '#427fda',
  },
});


const ProgressBar = ({ progress }) =>
  <View style={styles.progressView}>
    <View style={[styles.currentProgress, { flex: progress }]} />
    <View style={{ flex: 1 - progress }} />
  </View>;

ProgressBar.propTypes = {
  progress: PropTypes.number.isRequired,
};

export default ProgressBar;
