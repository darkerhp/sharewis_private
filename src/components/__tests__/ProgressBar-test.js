/* eslint-disable no-undef */
/* eslint-disable import/no-extraneous-dependencies */
import 'react-native';
import renderer from 'react-test-renderer';
import React from 'react';
import ProgressBar from '../ProgressBar';

describe('ProgressBar', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<ProgressBar progress={20} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
