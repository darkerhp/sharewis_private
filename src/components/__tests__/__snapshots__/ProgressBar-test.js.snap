exports[`ProgressBar renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#cdcdcd",
      "borderColor": "#cdcdcd",
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 18,
    }
  }>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#427fda",
        },
        Object {
          "flex": 20,
        },
      ]
    } />
  <View
    style={
      Object {
        "flex": -19,
      }
    } />
</View>
`;
